.github/workflows/snyk.yml

```diff
@@ -0,0 +1,31 @@
+name: Snyk
+
+on:
+  push:
+    branches: [ master ]
+
+jobs:
+  security:
+    runs-on: ubuntu-20.04
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup PHP
+      uses: shivammathur/setup-php@v2
+      with:
+        php-version: '7.2'
+        extensions: mbstring
+        coverage: xdebug
+
+    - name: Composer
+      run: |
+        composer self-update
+        composer install
+
+    - name: Run Snyk to check for vulnerabilities
+      uses: snyk/actions/php@master
```

Comment:
This uses a docker container under the hood, so your PHP set up above doesn't apply inside the container. If you want to use your set up (and not spin up a container), you might prefer to use the technique described here https://github.com/snyk/actions#bring-your-own-development-environment

Or you could keep just the snyk php action if it works as expected for this repo (it usually does but some times things are weird).
https://github.com/snyk/actions/tree/master/php#snyk-php-action

Reply:
Yes, indeed the PHP environment won't apply in the container. The reason i added those previous steps was in order to run Composer to create the `composer.lock` file. From Snyk [docs](https://support.snyk.io/hc/en-us/articles/360003817397-Snyk-for-PHP) i got the impression that the `composer.lock` file is required to be present for Snyk to generate the dependency graph. After looking at it again, it still seems the case (e.g. [here](https://github.com/snyk/snyk/issues/643#issuecomment-508900671), also `composer.json` is not in the detectable files [here](https://github.com/snyk/snyk/blob/master/src/lib/detect.ts)). It feels i may be missing something though! (i'll remove those steps and just let it run on this branch too to test)

Reply:
Shouldn't we commit `composer.lock`? Typically you would commit the lock file so you get the same dependencies on all machines.

Reply:
Indeed, this seems the best to do (and solves for Snyk as well). Thank you for the right direction!